elasticsearch/src/main/scala/com/snowplowanalytics/stream/loader/clients/ElasticsearchBulkSender.scala

```diff
@@ -0,0 +1,166 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.stream.loader
+package clients
+
+// AWS
+import com.amazonaws.services.kinesis.connectors.elasticsearch.ElasticsearchObject
+import com.amazonaws.auth.AWSCredentialsProvider
+
+// Java
+import com.google.common.base.Charsets
+import com.google.common.io.BaseEncoding
+import org.slf4j.LoggerFactory
+
+// Scala
+import com.sksamuel.elastic4s.ElasticsearchClientUri
+import com.sksamuel.elastic4s.http.{HttpClient, NoOpHttpClientConfigCallback}
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import org.apache.http.{Header, HttpHost}
+import org.apache.http.message.BasicHeader
+import org.elasticsearch.client.{RestClient, RestClientBuilder}
+import scala.util.{Failure => SFailure, Success => SSuccess}
+import org.json4s.jackson.JsonMethods._
+
+// Concurrent
+import scalaz.concurrent.Strategy
+import scala.concurrent.ExecutionContext.Implicits.global
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Tracker
+import com.snowplowanalytics.snowplow.scalatracker.Tracker
+
+class ElasticsearchBulkSender(
+  endpoint: String,
+  port: Int,
+  ssl: Boolean,
+  region: String,
+  awsSigning: Boolean,
+  username: Option[String],
+  password: Option[String],
+  override val maxConnectionWaitTimeMs: Long = 60000L,
+  override val maxAttempts: Int = 6,
+  documentType: String,
+  credentialsProvider: AWSCredentialsProvider,
+  override val tracker: Option[Tracker] = None
+) extends BulkSender[EmitterJsonInput] {
+  require(maxAttempts > 0)
+  require(maxConnectionWaitTimeMs > 0)
+
+
+  override val log = LoggerFactory.getLogger(getClass)
+
+  private val uri = ElasticsearchClientUri(s"elasticsearch://$endpoint:$port?ssl=$ssl")
+  private val httpClientConfigCallback =
+    if (awsSigning) new SignedHttpClientConfigCallback(credentialsProvider, region)
+    else NoOpHttpClientConfigCallback
+
+  private val formedHost = new HttpHost(endpoint, port,if (uri.options.getOrElse("ssl", "false") == "true") "https" else "http")
+  private val restClientBuilder = RestClient.builder(formedHost)
+    .setHttpClientConfigCallback(httpClientConfigCallback.asInstanceOf[RestClientBuilder.HttpClientConfigCallback])
+
+  if (username.orNull != null && password.orNull != null) {
+    val userpass = BaseEncoding.base64().encode(s"${username.get}:${password.get}".getBytes(Charsets.UTF_8))
+    val headers:Array[Header] = Array(new BasicHeader("Authorization", s"Basic $userpass"))
+    restClientBuilder.setDefaultHeaders(headers)
+  }
+
+  private val client: HttpClient = HttpClient.fromRestClient(restClientBuilder.build())
```

Comment:
why was the client instantiation changed?

Reply:
a mistake! reverting this part.